src/common/transformations/src/transformations/sdpa_to_paged_attention/state_management_pattern.cpp

```diff
@@ -333,6 +339,13 @@ ov::pass::StateManagementPattern::StateManagementPattern(ParameterVector& kv_par
                                                                           max_context_len.get_node_shared_ptr()};
         params.insert(params.end(), additional_params.begin(), additional_params.end());
 
+        if (use_block_indices_inputs) {
+            auto block_indices =
+                setName(std::make_shared<v0::Parameter>(element::i32, PartialShape{-1}), "block_indices");
+            params.insert(params.begin() + 7, block_indices);
```

Comment:
params.begin() + 7??
why not params.push_back(block_indices)?

Reply:
Because it cannot be randomly put in the parameters list. It has to be at the exact place

Reply:
agreed to refactor it later